src/pages/en/reference/directives-reference.md

```diff
@@ -213,7 +213,9 @@ const message = "Astro is awesome!";
 ```
 
 :::caution
-Using `define:vars` on a `<script>` or `<style>` tag implies the [`is:inline` directive](#isinline), which means your scripts or styles won't be bundled and will be inlined directly into the HTML.
+Using `define:vars` on a `<script>` or `<style>` tag implies the [`is:inline` directive](#isinline), which means your scripts or styles won't be bundled and will be inlined directly into the HTML. 
+
+This is because when Astro bundles scripts, it creates one bundle for all instances of the script on the page and runs that once. `define:vars` requires a script to rerun with each set of values, so Astro creates an inline script instead.
```

Comment:
Thanks for this Dan! I wonder if we can make this explanation a little more concrete (bundles, instances, etc. are a bit vague in my mind). Not entirely happy, but maybe we this could be a starting point?

```suggestion
This is because when Astro bundles a script, it includes and runs the script once even if you include the component containing the script multiple times on one page. `define:vars` requires a script to rerun with each set of values, so Astro creates an inline script instead.
```

Reply:
Thanks @delucis, definitely prefer this way of explaining it - will see if I can make it more concise